RecoLocalCalo/HGCalRecProducers/plugins/HGCalClusterTestProducer.cc

```diff
@@ -174,22 +188,31 @@ void HGCalClusterTestProducer::produce(edm::Event& evt,
     edm::Ptr<reco::BasicCluster> ptr(clusterHandle,i);
     clusterPtrs.push_back(ptr);
   }
-  
-  for( unsigned i = 0; i < clusterHandleSharing->size(); ++i ) {
-    edm::Ptr<reco::BasicCluster> ptr(clusterHandleSharing,i);
-    clusterPtrsSharing.push_back(ptr);
+
+  if(doSharing){
+    for( unsigned i = 0; i < clusterHandleSharing->size(); ++i ) {
+      edm::Ptr<reco::BasicCluster> ptr(clusterHandleSharing,i);
+      clusterPtrsSharing.push_back(ptr);
+    }
   }
 
   std::unique_ptr<std::vector<reco::HGCalMultiCluster> > 
     multiclusters( new std::vector<reco::HGCalMultiCluster> ), 
     multiclusters_sharing( new std::vector<reco::HGCalMultiCluster> );
 
-  *multiclusters = std::move(multicluster_algo->makePreClusters(clusterPtrs));
-  *multiclusters_sharing = std::move(multicluster_algo->makePreClusters(clusterPtrsSharing));
-  
+  struct timeval now;
+  struct timeval then;
+  gettimeofday(&then,0);
```

Comment:
May want to replace old, non-portable `gettimeofday` with `std::chrono` functions, see e.g. http://www.cplusplus.com/reference/chrono/high_resolution_clock/now/